kategory/src/test/kotlin/kategory/data/Function1Test.kt

```diff
@@ -0,0 +1,25 @@
+package kategory
+
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.matchers.shouldBe
+import org.junit.runner.RunWith
+
+@RunWith(KTestJUnitRunner::class)
+class Function1Test : UnitSpec() {
+    init {
+        testLaws(MonadLaws.laws(Function1.monad<Int>(), object : Eq<HK<Function1.F, Int>> {
+            override fun eqv(a: HK<Function1.F, Int>, b: HK<Function1.F, Int>): Boolean =
+                    a.ev()(1) == b.ev()(1)
+        }))
+
+        "Function1Monad.binding should for comprehend over all values of multiple Function0" {
```

Comment:
We may be able to add generic tests for Monad and Applicative to tests comprehensions and the applicative builder works for all monads and applicative instances instead of hardcoding these on each datatype test.